scripts/extensions/sams/src/api/sets.ts

```diff
@@ -0,0 +1,74 @@
+// Types
+import {ISuperdesk, IRestApiResponse} from 'superdesk-api';
+import {ISetItem} from '../interfaces';
+
+const RESOURCE = 'sams/sets';
+
+export function getAllSets(superdesk: ISuperdesk): Promise<Array<ISetItem>> {
+    const {gettext} = superdesk.localization;
+    const {notify} = superdesk.ui;
+
+    return superdesk.dataApi.query<ISetItem>(
+        RESOURCE,
+        1,
+        {field: 'name', direction: 'ascending'},
+        {},
+    )
+        .then((response: IRestApiResponse<ISetItem>) => {
+            return response?._items ?? [];
+        })
+        .catch((error: any) => {
+            notify.error(gettext('Failed to load all sets'));
+
+            return Promise.reject(error);
+        });
+}
+
+export function createSet(superdesk: ISuperdesk, item: Partial<ISetItem>): Promise<ISetItem> {
+    const {gettext} = superdesk.localization;
+    const {notify} = superdesk.ui;
+
+    return superdesk.dataApi.create<ISetItem>(RESOURCE, item)
+        .then((set: ISetItem) => {
+            notify.success(gettext('Set created successfully'));
+
+            return set;
+        })
+        .catch((error: any) => {
+            notify.error(gettext('Failed to create the Set'));
+
+            return Promise.reject(error);
+        });
+}
+
+export function updateSet(superdesk: ISuperdesk, original: ISetItem, updates: Partial<ISetItem>): Promise<ISetItem> {
+    const {gettext} = superdesk.localization;
+    const {notify} = superdesk.ui;
+
+    return superdesk.dataApi.patch<ISetItem>(RESOURCE, original, updates)
+        .then((set: ISetItem) => {
+            notify.success(gettext('Set updated successfully'));
+
+            return set;
+        })
+        .catch((error: any) => {
+            notify.error(gettext('Failed to update the Set'));
+
+            return Promise.reject(error);
```

Comment:
is this different than just `throw error`?

Reply:
I originally had this throw an error, but changed it to use `Promise.reject` after a peer review

Reply:
@pablopunk errors are for crash-like conditions which signify that something is wrong with the code and it should be logged. Validation issues are expected application behaviour.

Reply:
I was just asking, but now that you mention it, you're right on that but that doesn't have to do with `throw` vs `reject`. Also in this case the error could be something else other than a validation error, like a network error, and it will be catched by this block too

Reply:
Validation issues were being logged as errors in the console, that's why I asked Mark not to throw it. I agree that we could put a conditional here and throw errors for failures which are not because of validation.

Reply:
There are tickets in the pipeline to improve the validation vs errors, from the SAMS API right through to this front end. We can improve this handling with those tickets